webview/WebView2_spec.md

```diff
@@ -0,0 +1,247 @@
+{
+    // Important to validate that the Uri is what we expect from that webview.
+    string uriAsString = sender.UriSource.ToString();
+
+    if (args.Source == uriAsString)
+    {
+        HandleWebMessageAsString(args.WebMessageAsString);
+        HandleWebMessageAsJson(args.WebMessageAsJson);
+    }
+
+    else
+    {
+        // If the source is not validated, don't process the message.
+        return;
+    }
+};
+```
+<br>
+
+**NavigationStarting:**
+Occurs before the WebView2 navigates to new content. You can cancel navigation in a handler for this event by setting the WebView2NavigationStartingEventArgs.Cancel property to true.
+
+
+```
+myWebView.NavigationStarting += myWebView_NavigationStarting;
+
+private void myWebView_NavigationStarting(WebView2 sender, WebView2NavigationStartingEventArgs args)
+{
+    if (!IsAllowedUri(args.Uri))
+    {
+        args.Cancel = true;
+    }
+}
+```
+<br>
+
+**ContentLoading:**
+ContentLoading fires when new content has started loading on the webview's main frame or if a same page navigation occurs (such as through fragment navigations or history.pushState navigations). This follows the NavigationStarting event and precedes the NavigationCompleted event.
+
+<br>
+
+**SourceChanged:**
+Occurs when the 'Source' property is changed
+
+<br>
+
+**NavigationCompleted**:
+Occurs when the WebView2 has finished loading the current content or if the navigation has failed. To determine whether the navigation has failed, check the IsSuccess and WebErrorStatus properties of the WebViewNavigationCompletedEventArgs class.
+
+
+```
+myWebView2.NavigationCompleted += myWebView2_NavigationCompleted;
+
+private void myWebView2_NavigationCompleted(WebView2 sender, WebView2NavigationCompletedEventArgs args)
+{
+    if (args.IsSuccess == true)
+    {
+        statusTextBlock.Text = "Navigation to “ +
+        args.Uri.ToString() + " completed successfully.";
+    }
+
+    else
+    {
+    statusTextBlock.Text = "Navigation to: " + args.Uri.ToString() + " failed with error ” + args.WebErrorStatus,ToString();
+    }
+}
+```
+
+<br>
+
+**Reload:**
+Fires navigation event without altering the navigation stack.
+
+<br>
+
+# Remarks
+
+// This API may ship before the WinRT CoreWebView object is included, meaning functionality of the Webview2 will be limited to the APIs included on this object.
+
+# API Notes
+
+**Properties:**
+<br>
+Source – Gets or sets the Uniform Resource Identifier (URI) source of the HTML content to display in the WebView2 control.
+
+CoreWebView2 – Stores the core WebView object with full access to Anaheim WebView API set.
+<br>
+
+**Methods:**
+<br>
+void NavigateToString(String htmlContent) – Loads the specified HTML content as a new document.
+
+Windows.Foundation.IAsyncOperation ExecuteScriptAsync(String javascriptCode) –
+
+Executes the specified script function from the currently loaded HTML, with specific arguments, as an asynchronous action.
```

Comment:
>function [](start = 30, length = 8)

Nit: it doesn't have to be a function, just some JS code . If we remove 'function', then 'with specific arguments' should also be removed